zephyrproject-rtos/zephyr
# Bluetooth: Mesh: Add database for managing nodes and keys

This PR relates to issue https://github.com/zephyrproject-rtos/zephyr/issues/20259.

This is currently somewhat of a PoC but it's easier to discuss with some real code.

I have moved the the storage of nodes from `struct bt_mesh_net`, which is used for the local node, into a new `struct bt_mesh_cdb`.

The configuration database is needed by stack in the following cases:
- When provisioning a new node (local and remote device)
- When using the `cfg_cli` or any other model that needs device keys

There are probably more but those are the ones used right now I believe.

To test the sample I have the following setup:
* One board with `samples/bluetooth/mesh_provisioner`
* One board with `test/bluetooth/mesh_shell`

The `mesh_provisioner` should provision and configure itself and then wait for an unprovisioned beacon to trigger provisioning and configuration of the new node.

Run the following commands on the `mesh_shell` device:
```
mesh init
mesh uuid 112233445566778899
mesh pb-adv on
```

The `mesh_provisioner` should then:
* Provision a new node
* Add an app key
* Bind the app key to the health service
* Configure the health service to publish every 10 seconds

If all goes well, you should see something like the following every 10s:
```
Health Current Status from 0x0002
Health Test ID 0x00 Company ID 0x05f1: no faults
```
